### PR TITLE
Add MangaType to graphql

### PR DESCRIPTION
**Pull request by toyhammered**
I feel like MediaUnitInterface sounds weird...

### CODE REVIEW

File: app/graphql/types/chapter.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class Types::Chapter < Types::BaseObject
+  implements Types::MediaUnitInterface
```
Comment (NuckChorris) [inline app/graphql/types/chapter.rb]: Despite my stupid ass using `EpisodicInterface`, GraphQL apparently doesn't recommend the suffix of `Interface` so this should probably be `Types::MediaUnit`

File: app/graphql/types/chapter.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  description 'A chapter of a Manga'
+
+  field :id, ID, null: false
+
+  field :volume_number, Integer,
+    null: false,
+    description: 'The volume this chapter is related to'
```
Comment (NuckChorris) [inline app/graphql/types/chapter.rb:9]: `volume_number` is unused now that we have a Volume model.  Better to expose `field :volume, Types::Volume` instead, and let API users just access `volume { number }`

File: app/graphql/types/chapter.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+  field :published, GraphQL::Types::ISO8601DateTime,
```
Comment (NuckChorris) [inline app/graphql/types/chapter.rb]: Could you change this to `released_at`?  You can pass `method: :published` to map it to the published field in the db

File: app/graphql/types/manga.rb [hunk 4]
```diff
@@ -1,3 +1,27 @@
...
+    description 'Chapters for this manga'
+    argument :number, [Integer], required: false
+  end
+
+  def chapters(number: nil)
+    if number
+      object.chapters.where(number: number)
+    else
+      AssociationLoader.for(Manga, :chapters).load(object).then(&:to_a)
+    end
```
Comment (NuckChorris) [inline app/graphql/types/manga.rb:23]: :ok_hand: This whole implementation looks right to me so far

File: app/graphql/types/chapter.rb (hunk 2 above)
Comment (toyhammered) [reply app/graphql/types/chapter.rb:9]: sure, will add this.

File: app/graphql/types/volume.rb [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    }
+  end
+
+  field :number, Integer,
+    null: false,
+    description: 'The sequence number of this unit'
+
+  field :thumbnail, Types::Image,
+    null: true,
+    description: 'A thumbnail image for the unit'
```
Comment (NuckChorris) [inline app/graphql/types/volume.rb]: You've got some references to `unit` throughout here but it's volume not unit

File: app/graphql/types/volume.rb [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+  field :chapters_count, Integer,
+    null: false,
+    description: 'Total chaters per Volume'
```
Comment (NuckChorris) [inline app/graphql/types/volume.rb]: *chapters

File: app/graphql/types/volume.rb [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+  field :isbn, String,
+    null: false,
+    description: 'Unique Identification for this Volume'
```
Comment (NuckChorris) [inline app/graphql/types/volume.rb]: 1. This is an array of ISBN numbers (so `field :isbn, [String]`)
2. The description doesn't help much (they're ISBN numbers so they actually identify a specific release of the volume, but the description doesn't explain that)

**Review by NuckChorris — CHANGES_REQUESTED** [review]

File: app/graphql/types/volume.rb (hunk 7 above)
Comment (toyhammered) [reply app/graphql/types/volume.rb]: well thank you for clarifying that I really wasn't sure how to explain it tbh.

Comment (NuckChorris) [reply app/graphql/types/volume.rb]: Yeah ISBN numbers are a bit weird.  We don't use them internally to identify things, but they're useful for things like finding physical copies for sale

File: app/graphql/types/chapter.rb (hunk 2 above)
Comment (NuckChorris) [reply app/graphql/types/chapter.rb:9]: Could you remove the `volume_number` field entirely?  I want to remove some of the older stuff in the switch to GraphQL

File: app/graphql/types/volume.rb [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+    null: false,
+    description: 'The sequence number of this volume'
+
+  field :thumbnail, Types::Image,
+    null: true,
+    description: 'A thumbnail image for this volume'
+
+  field :chapters_count, Integer,
+    null: false,
+    description: 'Total chapters per volume'
```
Comment (NuckChorris) [inline app/graphql/types/volume.rb:27]: This makes it sound like a Manga-level attribute vs a volume-level thing.  How about this?

```suggestion
    description: 'Total chapters in this volume'
```

**Review by NuckChorris — CHANGES_REQUESTED** [review]

**Comment by toyhammered** [comment]
Already done by @matthewdias

**Pull request closed without merging**